zerver/tests/test_home.py

```diff
@@ -311,83 +311,55 @@ def test_home_demo_organization(self) -> None:
         self.assertEqual(set(actual_keys), set(expected_keys))
 
     def test_logged_out_home(self) -> None:
-        # Redirect to login on first request.
-        result = self.client_get("/")
-        self.assertEqual(result.status_code, 302)
-        self.assertEqual(result.url, "/login/")
-
-        # Tell server that user wants to log in anonymously
-        # Redirects to load webapp.
-        realm = get_realm("zulip")
-        result = self.client_post("/", {"prefers_web_public_view": "true"})
-        self.assertEqual(self.client.session.get("prefers_web_public_view"), True)
-        self.assertEqual(realm.enable_spectator_access, True)
-        self.assertEqual(result.status_code, 302)
-        self.assertEqual(result.url, "http://zulip.testserver")
-
-        # Disable spectator login. Since Realm.enable_spectator_access
-        # is False, the login should fail.
-        realm.enable_spectator_access = False
-        realm.save()
-
-        result = self.client_post("/", {"prefers_web_public_view": "true"})
-        self.assertEqual(self.client.session.get("prefers_web_public_view"), True)
-        self.assertEqual(realm.enable_spectator_access, False)
-        self.assertEqual(result.status_code, 302)
-        self.assertEqual(result.url, "/login/")
-
-        # Enable spectator login.
-        realm.enable_spectator_access = True
-        realm.save()
-
-        result = self.client_post("/", {"prefers_web_public_view": "true"})
-        self.assertEqual(self.client.session.get("prefers_web_public_view"), True)
-        self.assertEqual(realm.enable_spectator_access, True)
-        self.assertEqual(result.status_code, 302)
-        self.assertEqual(result.url, "http://zulip.testserver")
-
-        # Always load the web app from then on directly
-        result = self.client_get("/")
-        self.assertEqual(result.status_code, 200)
+        with patch.object(
+            Realm, "allow_web_public_streams_access"
+        ) as allow_web_public_streams_access:
+            allow_web_public_streams_access.return_value = False
```

Comment:
We shouldn't use this sort of mock strategy in tests -- we always want to actually change the configuration to disable web-public-streams, because that ends up being much more robust. (Depending on the test, we can do that with `with self.settings` at the server level, or `do_set_realm_property` at the realm level).